mparray/special/_special.py

```diff
@@ -65,6 +70,10 @@ def betaln(x, y):
 
 @vectorize
 def betainc(a, b, x):
+    if x < 0 or x > 1:
+        # The mpmath betainc implementation is defined on entire real line,
+        # (and the complex plane). We want to match scipy.special.betainc.
+        return mp.nan
```

Comment:
Hmm. I suppose when writing the standard we should have a minimum standard for the domain and what to do out-of-domain, but is there harm for extending the domain? Perhaps there are different extensions, and we'd have to define which to follow?

Reply:
I can’t remember exactly why I restricted the domain and am not at my computer now to check. I know I did it after implementing `betaincinv`, which isn’t included in this PR. We can drop the domain restriction here I think.